lib/LogCollector.js

```diff
@@ -0,0 +1,231 @@
+/* Copyright @ 2016 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+var Logger = require('./Logger.js');
+
+/**
+ * Stringifies given <tt>object</tt> without any circular references.
+ * @param {object} someObject the <tt>object</tt> to be stringified.
+ * @return {string} the result of <tt>JSON.stringify</tt> with the circular
+ * references skipped.
+ */
+function stringify(someObject){
+    const cache = [];
+    return JSON.stringify(someObject, function (key, value) {
```

Comment:
The problem is quite complex and is should probably be solved when the "stringificator" has more knowledge about the object's structure.
I've added a special string which prints object's index from the cache list which allows to potentially identify objects in the structure. Given that the purpose of this stringification is mostly debugging I'd rather not go further than that: 
https://github.com/jitsi/jitsi-meet-logger/pull/1/commits/3e8e2bd147329784b0f1ee1b6d2898dbe444bd08